numpy/_build_utils/gitversion.py

```diff
@@ -0,0 +1,83 @@
+import os
+import textwrap
+
+
+def init_version():
+    init = os.path.join(os.path.dirname(__file__), '../../pyproject.toml')
+    data = open(init).readlines()
+
+    version_line = next(
+        line for line in data if line.startswith('version =')
+    )
+
+    version = version_line.strip().split(' = ')[1]
+    version = version.replace('"', '').replace("'", '')
+
+    return version
+
+
+def git_version(version):
+    if 'dev' in version:
+        # Append last commit date and hash to dev version information,
+        # if available
+
+        import subprocess
+        import os.path
+
+        try:
+            p = subprocess.Popen(
+                ['git', 'log', '-1', '--format="%H %h %aI"'],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                cwd=os.path.dirname(__file__),
+            )
+        except FileNotFoundError:
+            pass
+        else:
+            out, err = p.communicate()
+            if p.returncode == 0:
+                git_hash, git_hash_short, git_date = (
+                    out.decode('utf-8')
+                    .strip()
+                    .replace('"', '')
+                    .split('T')[0]
+                    .replace('-', '')
+                    .split()
+                )
+
+                version__ = '+'.join(
+                    [tag for tag in version.split('+')
+                     if not tag.startswith('git')]
+                )
+                version += f'+git{git_date}.{git_hash_short}'
```

Comment:
I think we should be explicit how many characters to use
```suggestion
                version += f'+git{git_date}.{git_hash[:9]}'
```